mcts.py

```diff
@@ -238,10 +238,12 @@ def is_done(self):
         return self.position.is_game_over() or self.position.n >= FLAGS.max_game_length
 
     def inject_noise(self):
-        dirch = np.random.dirichlet(
-            [FLAGS.dirichlet_noise_alpha] * ((go.N * go.N) + 1))
+        epsilon = 1e-5
+        legal_moves = (1 - (self.illegal_moves / 1000)) + epsilon
+        a = legal_moves * ([FLAGS.dirichlet_noise_alpha] * (go.N * go.N + 1))
```

Comment:
I think this works too and is cleaner: `dirichlet = np.random.dirichlet(legal_moves * FLAGS.dirichlet_noise_alpha)`